src/components/CartButtons.js

```diff
@@ -3,33 +3,31 @@ import { FaShoppingCart, FaUserMinus, FaUserPlus } from 'react-icons/fa';
 import { Link } from 'react-router-dom';
 import styled from 'styled-components';
 import { useProductsContext } from '../context/products_context';
+import { useUserContext } from '../context/user_context';
 
 //vart-btn-wrapper- global class- see in Navbar.js it is display none on a default screen, nested class in NavContainer = styled...
 const CartButtons = () => {
-  console.log('RENDERING CART BUTTONS');
   const { closeSidebar } = useProductsContext(); // extracting the closeSidebar function from the returned object from useProductsContext() and assigning it to a variable named closeSidebar.
-  const [isLoginVisible, setIsLoginVisible] = useState(true);
-  console.log('isLoginVisible', isLoginVisible);
-
-  const handleLoginClick = () => {
-    setIsLoginVisible(false);
-  };
+  const { currentUser } = useUserContext();
```

Comment:
By calling `useUserContext` we can access variables that were defiend all the way up in the `user_context.js` component